File: src/python/learner.py
#!/usr/bin/python

from messenger import Messenger
from os        import getpid

messenger = Messenger('http://127.0.0.1:24192/l/dt-py-' + str(getpid()).zfill(5))
received = []

while True:
  currMessage = messenger.getNextMessage()
  for prevMessage in received:
    if currMessage['timePeriod'] == prevMessage['timePeriod'] and currMessage['by'] != prevMessage['by']:
      print "Value '" + currMessage['value'] + "' learned"
      break
  received.append(currMessage)

File: src/ruby/learner.rb
#!/usr/bin/ruby

require_relative 'messenger'

messenger = Messenger.new('http://127.0.0.1:24192/l/dt-rb')
received = []

while true do

  currMessage = messenger.getMessage

  for prevMessage in received do
    if currMessage['timePeriod'] != prevMessage['timePeriod'] then next end
    if currMessage['by'] == prevMessage['by'] then next end

    print "Value learned: '" + currMessage.to_s + "'\n"
    break
  end

  received << currMessage

end
